app/code/core/Mage/Core/Model/Encryption.php

```diff
@@ -81,25 +95,36 @@ public function getHash($password, $salt = false)
             $salt = $this->_helper->getRandomString($salt);
         }
         return $salt === false
-            ? $this->hash($password)
+            ? $this->hash($password, self::HASH_VERSION_SHA256)
             : $this->hash($salt . $password, self::HASH_VERSION_SHA256) . ':' . $salt;
     }
 
     /**
      * Generate hash for customer password
      *
      * @param string $password
-     * @param mixed $salt
+     * @param mixed $salt deprecated
      * @return string
      */
     public function getHashPassword($password, $salt = null)
     {
-        if (is_integer($salt)) {
-            $salt = $this->_helper->getRandomString($salt);
+        return $this->hash($password, self::HASH_VERSION_LATEST);
+    }
+
+    /**
+     * Generate hash for customer password
```

Comment:
Comment outdated? Does not reflect return type. Remove?

Reply:
aye thx, there were more wrong... I fixed them...